### PR TITLE
fix: restrict how many requests are made by app to api when no sql is…

### PR DESCRIPTION
**Pull request by MildTomato**
… present for logs request

## What kind of change does this PR introduce?

Stops a request to the API when there is no SQL ready to retrieve logs.

## What is the current behavior?

A request is made to API too soon, when the constructed SQL is not ready.
SWR then tries to refetch again once the SQL comes through which results in wasted requests to API.

## What is the new behavior?

Request is only made when SQL and raw SQL are defined

## Additional context

Add any other context or screenshots.

### CODE REVIEW

File: studio/hooks/analytics/useLogsPreview.tsx [hunk 1]
```diff
@@ -61,6 +61,13 @@ function useLogsPreview(
   // handle url generation for log pagination
   const getKeyLogs: SWRInfiniteKeyLoader = (_pageIndex: number, prevPageData: Logs) => {
     let queryParams
+
+    // check that SQL is ready for the logs request
+    if(!params.sql || !params.rawSql) {
```
Comment (Ziinc) [inline studio/hooks/analytics/useLogsPreview.tsx:66]: we can cut it down to:
```js
// cancel request if no sql provided
if (!params.sql) return null
```
only `sql` param is used in LF. also, useSWRInfinite uses `null` to cancel requests 

https://swr.vercel.app/docs/pagination#example-2-cursor-or-offset-based-paginated-api

File: studio/hooks/analytics/useLogsPreview.tsx [hunk 2]
```diff
@@ -86,13 +93,22 @@ function useLogsPreview(
...
-  const countUrl = `${API_URL}/projects/${projectRef}/analytics/endpoints/logs.all?${genQueryParams(
-    {
-      ...params,
-      sql: genCountQuery(table),
-      period_start: String(latestRefresh),
-    } as any
-  )}`
+  const countUrl = () => {
+    
+    // check that SQL is ready for the count request
```
Comment (Ziinc) [inline studio/hooks/analytics/useLogsPreview.tsx:98]: same as above

**Review by Ziinc — CHANGES_REQUESTED** [review]
🔥 thanks @MildTomato i was noticing the additional initial empty request but hadn't gotten round investigating it

File: studio/hooks/analytics/useLogsPreview.tsx (hunk 1 above)
Comment (MildTomato) [reply studio/hooks/analytics/useLogsPreview.tsx:66]: forgot to push up a change that changed it to null.
although I've cited the example to https://swr.vercel.app/docs/conditional-fetching#conditional in the comments

File: studio/hooks/analytics/useLogsPreview.tsx (hunk 2 above)
Comment (MildTomato) [reply studio/hooks/analytics/useLogsPreview.tsx:99]: done

**Review by kiwicopple — APPROVED** [review]

**Review by Ziinc — APPROVED** [review]

**Pull request merged**